qiskit/transpiler/passes/synthesis/unitary_synthesis.py

```diff
@@ -267,7 +274,7 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
                     self._coupling_map,
                     [dag_bit_indices[x] for x in node.qargs],
                 )
-            synth_dag = method.run(unitary, **kwargs)
+            synth_dag = method.run(unitary, config=self._plugin_config, **kwargs)
```

Comment:
This will pass the user-defined kwargs into the plugin, even if we're falling back to the default plugin. That won't _matter_ since the default one just ignores them, but it seems logically wrong - it feels like we should only be passing them to the "selected" plugin.  We could add `plugin_kwargs["config"] = self._plugin_config` and `default_kwargs["config"] = None` into the keyword processor step (possibly up at the top, where we sort out the case of the selected method being the default).

Reply:
Yeah, I was just being lazy. I'll update this now it's cleaner to do it this way

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7252/commits/a67482be0d503d82851c0c9b9b2f7f1a35930030